src/transformers/models/tvp/modeling_tvp.py

```diff
@@ -0,0 +1,914 @@
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ):
+        r"""
+        input_ids: (batch_size, sequence_length) pixel_values: (batch_size, frames, height, width, channels)
+        attention_mask: (batch_size, sequence_length) with 1 indicates valid, 0 indicates invalid position.
+        """
+        return_dict = return_dict if return_dict is not None else self.config.return_dict
+        input_shape = input_ids.size()
+        device = input_ids.device
+        # We can provide a self-attention mask of dimensions [batch_size, from_seq_length, to_seq_length]
+        # ourselves in which case we just need to make it broadcastable to all heads.
+        text_embedding_output = self.embeddings(input_ids=input_ids)  # (batch_size, sequence_length, hidden_size)
+        # (batch_size, visual_sequence_length, hidden_size)
+        visual_embedding_output = self.visual_embeddings(pixel_values)
+        # (batch_size, visual_sequence_length)
+        visual_attention_mask = attention_mask.new_ones(visual_embedding_output.shape[:2])
+        pt_mask = torch.ones(attention_mask.shape[0], 10)
+        attention_mask = torch.cat([pt_mask.long(), attention_mask, visual_attention_mask], dim=-1)
+        text_prompt = self.text_prompt.expand(text_embedding_output.shape[0], -1, -1)
+        # (batch_size, sequence_length + visual_sequence_length, hidden_size)
+        embedding_output = torch.cat([text_prompt, text_embedding_output, visual_embedding_output], dim=1)
+
+        extended_attention_mask = self.get_extended_attention_mask(attention_mask, input_shape).to(device)
+        encoder_outputs = self.encoder(
+            embedding_output,
+            attention_mask=extended_attention_mask,
+            head_mask=self.get_head_mask(head_mask, self.config.num_hidden_layers),
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+        last_hidden_state = encoder_outputs.last_hidden_state if return_dict else encoder_outputs[0]
+        pooled_output = self.pooler(last_hidden_state)
+        last_hidden_state = self.dropout(last_hidden_state)
+        pooled_output = self.dropout(pooled_output)
+        if not return_dict:
+            return (
+                last_hidden_state,
+                pooled_output,
+            ) + encoder_outputs[1:]
+
+        return BaseModelOutputWithPooling(
+            last_hidden_state=last_hidden_state,
+            pooler_output=pooled_output,
+            hidden_states=encoder_outputs.hidden_states,
+            attentions=encoder_outputs.attentions,
+        )
+
+
+# Pad frames extracted from videos only at the bottom.
+class TvpFrameDownPadPrompter(nn.Module):
+    def __init__(self, image_size, visual_prompt_size, frame_num, max_img_size, visual_prompter_apply):
```

Comment:
All custom layers should take the config as their first argument and take as few kwargs as possible. In this case, all these arguments can be removed